repos/system_upgrade/common/actors/repositoriesmapping/tests/unit_test_repositoriesmapping.py

```diff
@@ -1,114 +1,298 @@
+    is handled internally and StopActorExecutionError is propagated to the user.
+    """
+
+    json_data = {
+        'datetime': '202107141655Z',
+        'version_format': repositoriesmapping.RepoMapData.VERSION_FORMAT,
+        'mapping': [
+            {
+                'source_major_version': '7',
+                'target_major_version': '8',
+                'entries':  [
+                    {
+                        'source': 'source_pesid',
+                        'target': 'target_pesid'
+                    }
+                ]
+            }
+        ],
+        'repositories': [
+            {
+                'pesid': 'source_pesid',
+                'entries': [
+                    {
+                        "major_version": "7",
+                        "repoid": "some-rhel-9-repo1",
+                        "arch": "x86_64",
+                    }
+                ]
+            },
+            {
+                'pesid': 'target_pesid',
+                'entries': [
+                    {
+                        "major_version": "7",
+                        "repoid": "some-rhel-9-repo1",
+                        "arch": "x86_64",
+                    }
+                ]
+            }
+        ]
+    }
+
+    monkeypatch.setattr(api, 'current_actor', CurrentActorMocked())
     monkeypatch.setattr(api, 'produce', produce_mocked())
-    # add one comment and one empty line into the repomap file
-    repofile = ReadRepoFileMock(input_data)
-    repofile.repomap_file.insert(2, '')
-    repofile.repomap_file.insert(3, '# comment')
-    # run
-    repositoriesmapping.scan_repositories(read_repofile_func=repofile)
-    assert api.produce.called == 1
-    assert api.produce.model_instances == [gen_RepositoriesMap(expected_records)]
+
+    # RepositoriesMapping imports version checking functions all the way into its namespace
+    monkeypatch.setattr(repositoriesmapping, 'get_source_major_version', lambda: '7')
+    monkeypatch.setattr(repositoriesmapping, 'get_target_major_version', lambda: '8')
+
+    with pytest.raises(StopActorExecutionError) as error_info:
+        repositoriesmapping.scan_repositories(lambda dummy: json_data)
+
+    assert 'repository mapping file is invalid' in error_info.value.message
+
+
+# # @pytest.mark.parametrize('arch,src_type,dst_type', gen_input_permutation())
+# # def test_scan_valid_file_without_comments(monkeypatch, arch, src_type, dst_type):
+# #     envars = {'LEAPP_DEVEL_SOURCE_PRODUCT_TYPE': src_type, 'LEAPP_DEVEL_TARGET_PRODUCT_TYPE': dst_type}
+# #     input_data, expected_records = gen_test_data(arch, src_type, dst_type)
+# #     monkeypatch.setattr(api, 'current_actor', CurrentActorMocked(arch, envars))
+# #     monkeypatch.setattr(api, 'produce', produce_mocked())
+# #     repositoriesmapping.scan_repositories(read_repofile_func=ReadRepoFileMock())
+# #     assert api.produce.called == 1
+# #     assert api.produce.model_instances == [gen_RepositoriesMap(expected_records)]
```

Comment:
This is not relevant anymore, remove it.

Reply:
removed.